layers/timeline_semaphore.c

```diff
@@ -2135,6 +2136,8 @@ static VkResult timeline_EnumeratePhysicalDevices(
                 object_unmap(&global_map, pPhysicalDevices[i]);
             object_map(&global_map, pPhysicalDevices[i], instance);
         }
+
+        instance->physicalDeviceCount = *pPhysicalDeviceCount; 
```

Comment:
I would write : 
```
instance->physicalDeviceCount = MAX(instance->physicalDeviceCount, *physicalDeviceCount);
```

This could be called multiple times, but the current code only remembers the last number which could be lower than the biggest number seen in this call.

Reply:
Right!